sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_responses.hpp

```diff
@@ -9,15 +9,49 @@
 
 namespace Azure { namespace Storage { namespace Files { namespace Shares { namespace Models {
 
+  struct FileShareSmbProperties
+  {
+    /**
+     * @brief Key of the permission to be set for the directory/file.
+     */
+    Azure::Core::Nullable<std::string> PermissionKey;
+
+    /**
+     * @brief If specified, the provided file attributes shall be set. Default value:
+     * FileAttribute::Archive for file and FileAttribute::Directory for directory.
+     * FileAttribute::None can also be specified as default.
+     */
+    FileAttributes Attributes = static_cast<FileAttributes>(0);
```

Comment:
Should we have a `None` attribute?

Reply:
`static_cast<FileAttributes>(0)`

Is this what our customers are supposed to write code?

Reply:
Revisited what .NET does, we do have `None` at the moment but we shouldn't be initializing it to this value. I have made the member to `Azure::Core::Nullable<FileAttributes> Attributes;` to resolve this issue.

Reply:
so when `Attributes` will be null?

Reply:
is this `FileShareSmbProperties` struct used in return  types or options?

Reply:
As discussed with Ted, it is used in both return types and options. Customer doesn't have to specify it in options, but I think in return types it will always not be null.

Reply:
Well this is not good. If it's always non-null, it should not be nullable.

Reply:
Here's another question, when it's used in options, how will our customer specify `None` to clear all attributes, and `preserve`, and `inherit`?

Reply:
https://github.com/Azure/azure-sdk-for-net/blob/4f191fb32a70446efa6b1efc729692600c0e1587/sdk/storage/Azure.Storage.Files.Shares/src/ShareFileClient.cs#L716
In the REST layer it is required, but .NET makes it nullable and clears what the server stores if it is null just like other properties.

Reply:
what if the user wants  to use `preserve` or `inherit`?

Reply:
I think it is a feature missing in .NET SDK.
`inherit` is not available for the parameter so one thing we can do in C++ is that null means `preserve`. But we have the risk of future adding `inherit` to this support matrix. What do you think?

Reply:
I don't know

Reply:
Email sent, I'm proposing to add `preserve` as an attribute enum, let's see what others think.

Reply:
this will be resolved in another PR as it is not related here.

Reply:
Create an issue to track that.

Reply:
https://github.com/Azure/azure-sdk-for-cpp/issues/1305